EIPS/eip-3540.md

```diff
@@ -0,0 +1,218 @@
+---
+eip: 3540
+title: EVM Object Format (EOF) v1
+author: Alex Beregszaszi (@axic), Paweł Bylica (@chfast), Andrei Maiboroda (@gumb0)
+discussions-to: https://ethereum-magicians.org/t/evm-object-format-eof/5727/4
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-03-16
+---
+
+## Abstract
+
+We introduce an extensible and versioned container format for the EVM with a once-off validation at deploy time. The version described here brings the tangible benefit of code and data separation, and allows for easy introduction of a variety of changes in the future. It is introduced via two hard forks to avoid breaking any existing executable contracts.
+
+## Motivation
+
+On-chain deployed EVM bytecode contains no pre-defined structure today. Code is typically validated in clients to the extent of `JUMPDEST` analysis at runtime, every single time prior to execution. This poses not only an overhead, but also a challenge for introducing new or deprecating old features. [This initial proposal](https://notes.ethereum.org/@axic/evm-object-format) explains some of these challenges.
```

Comment:
External links are strongly discouraged.  Please inline the relevant content here rather than linking externally to it.

Alternatively, consider including it as an asset in `../assets/eip-3540/evm-object-format.pdf` or something.

Reply:
I disagree with this.